Physics/Dynamics/ProductionFormFactor.hpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2013, 2019 The ComPWA Team.
+// This file is part of the ComPWA framework, check
+// https://github.com/ComPWA/ComPWA/license.txt for details.
+
+#ifndef PHYSICS_DYNAMICS_PRODUCTION_FORMFACTOR_HPP_
+#define PHYSICS_DYNAMICS_PRODUCTION_FORMFACTOR_HPP_
+
+#include <boost/property_tree/ptree.hpp>
+#include <memory>
+#include <vector>
+
+#include "AbstractDynamicalFunction.hpp"
+#include "Core/Exceptions.hpp"
+#include "Core/Functions.hpp"
+#include "Core/Spin.hpp"
+#include "FormFactor.hpp"
+
+namespace ComPWA {
+namespace Physics {
+namespace Dynamics {
+
+/// The production formfactor is implemented in Blatt-Weisskopf type \f$B_{L}(q)\f$
+/// Ref. Dalitz Plot Analysis, PDG 2012, where 
+/// \f[
+/// \frac{B^{\prime}(q_{0})}{B^{\prime}_{L}(q)} = q^{L}B^{\prime}_{L}(q, q_{0})
+/// \f]
+/// \f$B_{L}(q)(i.e., F_L(q))$\f with \f$L$\f up to 4 can be found in BNL-QGS-06-101
+class ProductionFormFactor
+    : public ComPWA::Physics::Dynamics::AbstractDynamicalFunction {
+
+public:
+  //============ CONSTRUCTION ==================
+  ProductionFormFactor(std::string name, 
+                       std::string daug1Name, std::string daug2Name,
+                       std::shared_ptr<ComPWA::FitParameter> mass1, 
+                       std::shared_ptr<ComPWA::FitParameter> mass2,
+                       std::shared_ptr<ComPWA::FitParameter> radius,
+                       ComPWA::Spin orbitL,
+                       FormFactorType ffType);
+  virtual ~ProductionFormFactor();
+
+  //================ EVALUATION =================
+
+  std::complex<double> evaluate(const ComPWA::DataPoint &point,
+                                unsigned int pos) const;
+
+  /// Dynamical Blatt-Weisskopf function.
+  /// \param mSq Invariant mass squared
+  /// \param ma Mass of daughter particle
+  /// \param mb Mass of daughter particle
+  /// \param L Orbital angular momentum between two daughters a and b
+  /// \param mesonRadius Meson Radius
+  /// \param ffType Form factor type
+  static std::complex<double>
+  dynamicalFunction(double mSq, double ma, double mb, unsigned int L, 
+                    double mesonRadius, FormFactorType ffType);
+
+  //============ SET/GET =================
+
```

Comment:
Are these getters and setters really necessary? I think you can remove all of them.

Reply:
emm, you are right